File: WebApp/wwwroot/scripts/domain/voucher.ts
module App.Models {
    "use strict";

    export class Voucher {
        public id: string;
        public title: string;
        public code: string;
        public period: number;
        public isRedeemed: boolean;
        public paymentRecordId: string;
        public organisationId: string;
        public dateCreated: number;
        public dateUpdated: number;

        constructor() {
            this.id = '';
            this.title = '';
            this.code = '';
            this.period = 1;
            this.isRedeemed = false;
            this.paymentRecordId = '';
            this.organisationId = '';
            this.dateCreated = _.now();
            this.dateUpdated = _.now();
        }
    }

    export interface IVoucher extends ng.resource.IResource<IVoucher> {
        id: string;
        title: string;
        code: string;
        period: number;
        isRedeemed: boolean;
        paymentRecordId: string;
        organisationId: string;
        dateCreated: number;
        dateUpdated: number;
    }
}

File: Models/Entities/Voucher.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace LightMethods.Survey.Models.Entities
{
    public class Voucher : Entity
    {
        public string Title { get; set; }

        [Index("IX_Voucher_Code", 1, IsUnique = true)]
        public string Code { get; set; }

        public int Period { get; set; }

        public bool IsRedeemed { get; set; }

        public Guid? PaymentRecordId { get; set; }

        public virtual PaymentRecord PaymentRecord { get; set; }

        public Guid OrganisationId { get; set; }

        public virtual Organisation Organisation { get; set; }
    }
}
